tests/test_jumpy.py

```diff
@@ -0,0 +1,562 @@
+    """
+    A = onp.array([1, 2, 3])
+    B = jnp.array([1, 2, 3])
+
+    _check_types(A, jp.exp(A))
+    _check_types(B, jp.exp(B))
+
+
+def test_sign():
+    """
+    Calls the function `sign` on both `onp.array` and `jnp.array`
+    and checks the respective return types
+    """
+    A = onp.array([1, 2, 3])
+    B = jnp.array([1, 2, 3])
+
+    _check_types(A, jp.sign(A))
+    _check_types(B, jp.sign(B))
+
+
+def test_sum():
+    """
+    Calls the function `sum` on both `onp.array` and `jnp.array`
+    to check it doesn't raise an error
+    """
+    A = onp.array([1, 2, 3])
+    B = jnp.array([1, 2, 3])
+
+    jp.sum(A, axis=0)
+    jp.sum(A)
+    jp.sum(B, axis=0)
+    jp.sum(B)
+
+
+def test_stack():
+    """
+    Calls the function `stack` on both `onp.array` and `jnp.array`
+    and checks the respective return types
+    """
+    A = onp.array([1, 2, 3])
+    B = jnp.array([1, 2, 3])
+        
+    _check_types(A, jp.stack([A, A]))
+    _check_types(B, jp.stack([B, B]))
+    _check_types_multiple((A,B), jp.stack([A, B]))
+
+
+def test_concatenate():
+    """
+    Calls the function `concatenate` on both `onp.array` and 
+    `jnp.array` and checks the respective return types
+    """
+    A = onp.array([1, 2, 3])
+    B = jnp.array([1, 2, 3])
+        
+    _check_types(A, jp.concatenate([A, A]))
+    _check_types(B, jp.concatenate([B, B]))
+    _check_types_multiple((A,B), jp.concatenate([A, B]))
+
+
+def test_sqrt():
+    """
+    Calls the function `sqrt` on both `onp.array` and `jnp.array`
+    and checks the respective return types
+    """
+    A = onp.array([1, 2, 3])
+    B = jnp.array([1, 2, 3])
+
+    _check_types(A, jp.sqrt(A))
+    _check_types(B, jp.sqrt(B))
+
+
+def test_where():
+    """
+    Calls the function `where` on both `onp.array` and `jnp.array`
+    and checks the respective return types
+    """
+    A = onp.array([1, 2, 3])
+    B = onp.ones(3)
+    C = jnp.array([1, 2, 3])
+    D = jnp.ones(3)
+        
+    _check_types_multiple((A, B), jp.where(B, A, A))
+    _check_types_multiple((C, D), jp.where(D, C, C))
+    _check_types_multiple((A,B, C), jp.where(B, A, C))
+
+
+def test_diag():
+    """
+    Calls the function `diag` on both `onp.array` and `jnp.array`
+    and checks the respective return types
+    """
+    A = onp.arange(9).reshape((3,3))
+    B = jnp.arange(9).reshape((3,3))
+
+    _check_types(A, jp.diag(A))
+    _check_types(B, jp.diag(B))
+
+
+def test_clip():
+    """
+    Calls the function `clip` on both `onp.array` and `jnp.array`
+    and checks the respective return types
+    """
+    A = onp.array([1, 2, 3])
+    B = onp.ones(3)
+    C = jnp.array([1, 2, 3])
+    D = jnp.ones(3)
+        
+    _check_types_multiple((A, B), jp.where(B, A, A))
+    _check_types_multiple((C, D), jp.where(D, C, C))
+    _check_types_multiple((A,B, C), jp.where(B, A, C))
+
+
+def test_eye():
+    """
+    Calls the function `eye` to check it doesn't raise an error
+    """
+    jp.eye(3)
```

Comment:
Is this a jax or numpy function? Is it possible to get both currently?
Could we check that the output is the expected type

Reply:
The fact with those functions, if I understood correctly, is that `_which_np` gets called without any argument, resulting in always using standard numpy. So technically there would be no way to get a jax array from this... I don't know what the expected behaviour should be.

Reply:
Could we use a dtype parameter where users can use a jax dtype for jax array and numpy dtype for numpy array? @jkterry1 Thoughts?